addons/web/static/src/js/tools/debug_manager_backend.js

```diff
@@ -678,88 +704,59 @@ var RequestDetails = Widget.extend({
-    //         if (this.currentDialogController) {
-    //             return this.actions[this.currentDialogController.actionID];
-    //         }
-    //         return null;
-    //     },
-    //     /**
-    //      * Returns the controller currently opened in a dialog, if any.
-    //      *
-    //      * @returns {Object|null}
-    //      */
-    //     getCurrentControllerInDialog: function () {
-    //         return this.currentDialogController;
-    //     },
-    // });
-
-    Dialog.include({
-        //--------------------------------------------------------------------------
-        // Public
-        //--------------------------------------------------------------------------
-
-        /**
-         * @override
-         */
-        open: function() {
-            var self = this;
-            // if the dialog is opened by the ActionManager, instantiate a
-            // DebugManager and insert it into the DOM once the dialog is opened
-            // (delay this with a setTimeout(0) to ensure that the internal
-            // state, i.e. the current action and controller, of the
-            // ActionManager is set to properly update the DebugManager)
-            this.opened(function() {
-                setTimeout(function () {
-                    var parent = self.getParent();
-                    if (parent instanceof ActionManager) {
-                        var action = parent.getCurrentActionInDialog();
-                        if (action) {
-                            var controller = parent.getCurrentControllerInDialog();
-                            self.debugManager = new DebugManager(self);
-                            var $header = self.$modal.find('.modal-header:first');
-                            return self.debugManager.prependTo($header).then(function () {
-                                self.debugManager.update('action', action, controller.widget);
-                            });
-                        }
-                    }
-                }, 0);
+    // dialog (deployed in the header of action dialogs)
+    utils.patch(DialogAction, 'DialogActionDebug', {
+        async willStart() {
+            await this._super(...arguments);
+            this.debugManager = new DebugManagerAdapter(this, {
+                Component: DebugManager,
             });
-
-            return this._super.apply(this, arguments);
+            await this.debugManager.mount(document.createDocumentFragment());
+        },
+        mounted() {
+            this.dialog.comp.headerRef.el.prepend(this.debugManager.el);
+            this._super(...arguments);
         },
     });
+};
+DebugManager.undeployDebugManager = function () {
+    // main
+    const index = SystrayMenu.Items.indexOf(DebugManager);
+    if (index > -1) {
+        SystrayMenu.Items.splice(index, 1);
+    }
+
+    // dialog
+    utils.unpatch(DialogAction, 'DialogActionDebug');
+};
+
+if (config.isDebug()) {
```

Comment:
@aab-odoo 
Executing tests in debug mode makes that condition always true, and as expected, the route check access rights crashes.
Also, in boot.js, when setting odoo.testing, QUnit is not loaded yet. moving QUnit on top of the asset solves that particular problem, but makes another one.
From what I understand, Qunit sets its onUnhandledRejection before we had a chance to make our custom promise. This is only in the case qunit is set before all other assets